apollo-client/src/main/java/com/ctrip/framework/apollo/spi/DefaultConfigFactory.java

```diff
@@ -72,7 +72,9 @@ public Config create(String namespace) {
     // should not create a PropertiesCompatibleFileConfigRepository for them
     // calling the method `createLocalConfigRepository(...)` is more suitable
     // for ConfigFileFormat.Properties
-    if (ConfigFileFormat.isPropertiesCompatible(format) &&
+    if (!m_configUtil.isPropertyFileCacheEnabled()) {
```

Comment:
I think we could not add the logic here, as createConfigFile also will create local config repository.
How about we add another method, e.g. `createConfigRepository`, check whether the file cache enabled and create the correct config repository, then change all the places that call `createLocalConfigRepository` to `createConfigRepository`?

```java
  ConfigRepository createConfigRepository(String namespace) {
    if (m_configUtil.isPropertyFileCacheEnabled()) {
      return createLocalConfigRepository(namespace);
    }
    return createRemoteConfigRepository(namespace);
  }
```